src/librustc/ty/wf.rs

```diff
@@ -156,7 +175,7 @@ impl<'a, 'gcx, 'tcx> WfPredicates<'a, 'gcx, 'tcx> {
         // WF and (b) the trait-ref holds.  (It may also be
         // normalizable and be WF that way.)
         let trait_ref = data.trait_ref(self.infcx.tcx);
-        self.compute_trait_ref(&trait_ref);
+        self.compute_trait_ref(&trait_ref, Elaborate::All);
```

Comment:
I guess we could just have `Elaborate::None` here, since the implied obligations are proved once and for all when typechecking the impl declaration.